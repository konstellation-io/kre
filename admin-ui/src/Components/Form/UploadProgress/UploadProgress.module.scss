@import '../../../styles/mixins';
@import '../../../styles/colors';

.header {
  @include font-caption;

  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.uploaded {
  opacity: 0;
  color: $color-primary;

  &.show {
    opacity: 1;
  }
}

.progressContainer {
  position: relative;
  width: 100%;
  height: 2px;
  margin: 12px 0 16px 0;

  background-color: font-color(regular);
}
.progress {
  height: 100%;
  background-color: $color-primary;
}

.ack {
  @include font-small;
  font-weight: 400;
  color: $color-ok;

  opacity: 0;

  &.show {
    opacity: 1;
  }
}
