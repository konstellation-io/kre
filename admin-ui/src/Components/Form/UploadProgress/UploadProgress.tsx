import React from 'react';
import cx from 'classnames';
import styles from './UploadProgress.module.scss';

type Props = {
  fileName: string;
  progress: number;
};

function UploadProgress({ fileName, progress }: Props) {
  const fileUploaded = progress === 100;

  return (
    <>
      <div className={styles.header}>
        <span>PROGRESS</span>
        <span className={cx(styles.uploaded, { [styles.show]: fileUploaded })}>
          UPLOADED
        </span>
      </div>
      <div className={styles.progressContainer}>
        <div className={styles.progress} style={{ width: `${progress}%` }} />
      </div>
      <div className={cx(styles.ack, { [styles.show]: fileUploaded })}>
        {`The file "${fileName}" its ok, have no errors.`}
      </div>
    </>
  );
}

export default UploadProgress;
