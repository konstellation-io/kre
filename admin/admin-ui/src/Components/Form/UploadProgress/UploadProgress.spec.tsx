import React from 'react';
import { shallow } from 'enzyme';
import UploadProgress from './UploadProgress';

describe('UploadProgress component', () => {
  let wrapper;

  const fileNameMock = 'foo';
  const progressMock = 50;
  beforeEach(() => {
    jest.resetAllMocks();
    wrapper = shallow(
      <UploadProgress fileName={fileNameMock} progress={progressMock} />
    );
  });

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show fileUploaded if progress is 100', () => {
    wrapper.setProps({ progress: 100 });
    expect(wrapper).toMatchSnapshot();
  });
});
