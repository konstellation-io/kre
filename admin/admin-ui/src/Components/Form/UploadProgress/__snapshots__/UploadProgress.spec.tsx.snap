// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadProgress component should render without crashing 1`] = `
<Fragment>
  <div
    className="header"
  >
    <span>
      PROGRESS
    </span>
    <span
      className="uploaded"
    >
      UPLOADED
    </span>
  </div>
  <div
    className="progressContainer"
  >
    <div
      className="progress"
      style={
        Object {
          "width": "50%",
        }
      }
    />
  </div>
  <div
    className="ack"
  >
    The file "foo" its ok, have no errors.
  </div>
</Fragment>
`;

exports[`UploadProgress component should show fileUploaded if progress is 100 1`] = `
<Fragment>
  <div
    className="header"
  >
    <span>
      PROGRESS
    </span>
    <span
      className="uploaded show"
    >
      UPLOADED
    </span>
  </div>
  <div
    className="progressContainer"
  >
    <div
      className="progress"
      style={
        Object {
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="ack show"
  >
    The file "foo" its ok, have no errors.
  </div>
</Fragment>
`;
